src/NServiceBus.Core/Hosting/StartupDiagnostics/HostDiagnosticsConfigurationExtensions.cs

```diff
@@ -0,0 +1,42 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using Configuration.AdvancedExtensibility;
+
+    /// <summary>
+    /// Provides diagnostics configuration options.
+    /// </summary>
+    public static class HostDiagnosticsConfigurationExtensions
+    {
+        /// <summary>
+        /// Configures a custom path where host diagnostics is written.
+        /// </summary>
+        /// <param name="config">Configuration object to extend.</param>
+        /// <param name="path">The custom path to use.</param>
+        public static void SetDiagnosticsRootPath(this EndpointConfiguration config, string path)
```

Comment:
why only the root path? What if I don't want those files be in `.diagnostics`? Not sure this is really an issue, but I'd say usually I expect to be able to customize the whole path if I specify my own?

Reply:
Turns out it works as expected but name was misleading, renamed to `SetDiagnosticsPath`.